### PR TITLE
Scripting fixes; fix bugs; add REPL scripting loop 

### PR DESCRIPTION
**Pull request by baughj**
## Description
HYB-64: bugfixing
HYB-82: correct requirement for legend marks to have prefixes
HYB-18: /legend now allows dates / quantities to be properly set for testing, and doesn't make it so you can't login 😄 
HYB-36 / #145 - correctly handle primary/secondary/utility skills in panes
HYB-79: perform null checks for formula parser tokens that may be null

**Other**

Add slight-REPL loop for scripting usage, using whisper target $
Fix error in intent calculations: monsters should always be healable / statusable
Fix crash error in `e` (who list)
Fix null check error in weapon durability

### CODE REVIEW

File: hybrasyl/FormulaParser.cs [hunk 1]
```diff
@@ -271,51 +271,66 @@ public bool MapTokens(ref List<string> tokens)
...
                         case "$CASTERWEAPONLDMG":
                             {
                                 var rand = new Random();
-                                var mindmg = (int)_caster.Equipment.Weapon.MinLDamage;
-                                var maxdmg = (int)_caster.Equipment.Weapon.MaxLDamage;
-                                if (mindmg == 0) mindmg = 1;
-                                if (maxdmg == 0) maxdmg = 1;
-                                tokens[i] = rand.Next(mindmg, maxdmg).ToString();
+                                if (_caster.Equipment.Weapon == null)
+                                    tokens[i] = "0";
```
Comment (norrismiv) [inline hybrasyl/FormulaParser.cs:315]: attacking with no weapon equipped should have a 1m1 damage capacity, so this should read 1.

File: hybrasyl/FormulaParser.cs [hunk 2]
```diff
@@ -271,51 +271,66 @@ public bool MapTokens(ref List<string> tokens)
...
+                                    if (mindmg == 0) mindmg = 1;
+                                    if (maxdmg == 0) maxdmg = 1;
+                                    tokens[i] = rand.Next(mindmg, maxdmg).ToString();
+                                }
                             }
                             break;
                         case "$CASTERWEAPONLDMGMIN":
                             {
-                                tokens[i] = _caster.Equipment.Weapon.MinLDamage.ToString();
+                                tokens[i] = (_caster.Equipment.Weapon?.MinLDamage ?? '0').ToString();
```
Comment (norrismiv) [inline hybrasyl/FormulaParser.cs:328]: attacking with hands should be 1 base damage

File: hybrasyl/FormulaParser.cs [hunk 3]
```diff
@@ -271,51 +271,66 @@ public bool MapTokens(ref List<string> tokens)
...
                             }
                             break;
                         case "$CASTERWEAPONLDMGMAX":
                             {
-                                tokens[i] = _caster.Equipment.Weapon.MaxLDamage.ToString();
+                                tokens[i] = (_caster.Equipment.Weapon?.MaxLDamage ?? '0').ToString();
```
Comment (norrismiv) [inline hybrasyl/FormulaParser.cs:333]: attacking with hands should be 1 base damage

File: hybrasyl/Scripting/Script.cs [hunk 4]
```diff
@@ -144,6 +150,18 @@ public bool HasFunction(string name)
...
         }
 
+        public void SetGlobals()
+        {
+            // TODO: streamline / refactor
+            Compiled.Globals["Gender"] = UserData.CreateStatic<Xml.Gender>();
+            Compiled.Globals["LegendIcon"] = UserData.CreateStatic<LegendIcon>();
+            Compiled.Globals["LegendColor"] = UserData.CreateStatic<LegendColor>();
+            Compiled.Globals["Class"] = UserData.CreateStatic<Xml.Class>();
+            Compiled.Globals["utility"] = typeof(HybrasylUtility);
```
Comment (norrismiv) [inline hybrasyl/Scripting/Script.cs:160]: should this be lowercase?

File: hybrasyl/Scripting/Script.cs [hunk 5]
```diff
@@ -144,6 +150,18 @@ public bool HasFunction(string name)
...
+            Compiled.Globals.Set("world", UserData.Create(Processor.World));
```
Comment (norrismiv) [inline hybrasyl/Scripting/Script.cs:161]: see above

File: hybrasyl/Scripting/Script.cs [hunk 6]
```diff
@@ -144,6 +150,18 @@ public bool HasFunction(string name)
...
+            Compiled.Globals.Set("logger", UserData.Create(new ScriptLogger(Name)));
```
Comment (norrismiv) [inline hybrasyl/Scripting/Script.cs:162]: above

File: hybrasyl/Scripting/Script.cs (hunk 4 above)
Comment (baughj) [reply hybrasyl/Scripting/Script.cs:160]: no, type

File: hybrasyl/Scripting/Script.cs (hunk 5 above)
Comment (baughj) [reply hybrasyl/Scripting/Script.cs:161]: no, type

File: hybrasyl/Scripting/Script.cs (hunk 6 above)
Comment (baughj) [reply hybrasyl/Scripting/Script.cs:162]: no, type

**Comment by baughj** [comment]
no, type

**Pull request merged**